src/controllers/playback/video/index.js

```diff
@@ -1574,6 +1574,30 @@ import { setBackdropTransparency, TRANSPARENCY_LEVEL } from '../../../components
             return '<h1 class="sliderBubbleText">' + datetime.getDisplayRunningTime(ticks) + '</h1>';
         };
 
+        nowPlayingPositionSlider.getChapterNamesAndFractions = function () {
+            showOsd();
+
+            const item = currentItem;
+
+            if (item && item.Chapters && item.Chapters.length) {
+                const chapterNames = [];
+                const chapterFractions = [];
+                const runtimeDuration = item.RunTimeTicks;
+
+                for (let i = 0, length = item.Chapters.length; i < length; i++) {
+                    const currentChapter = item.Chapters[i];
+
+                    const fraction = currentChapter.StartPositionTicks / runtimeDuration;
+                    chapterFractions.push(fraction);
+                    chapterNames.push(currentChapter.Name);
```

Comment:
Agreed on returning an object instead of the "double array".

I changed all names to be more generic so that the `className` can be used to specify different kinds of markers, if ever needed. Does this new way fit your idea?